headers/assert_h.md

```diff
@@ -1,8 +1,63 @@
 # \<assert.h\>
 
-Reference: [\<assert.h\>](https://en.cppreference.com/w/c/assert)
+C Reference: [\<assert.h\>](https://en.cppreference.com/w/c/error)
 
-## Types
+Relevant Zig Documentation:
+  - [std.debug.assert](https://ziglang.org/documentation/master/std/#std;debug.assert)
 
-## Functions
+## [assert()](https://en.cppreference.com/w/c/error/assert)
 
+For the following C code, which will immediately abort the program if the condition is false. 
+
+```c
+double x = -1.0;
+assert(x >= 0.0);
+```
+
+**Equivalent Zig implementation:**
+
+```zig
+const assert = std.debug.assert;
+// ...
+const x: double = -1.0;
+assert(x >= 0.0); // undefined behaviour
+```
+
+This will invoke an undefined behaviour when the condition is `false`. Checkout [std.debug.assert](https://ziglang.org/documentation/master/std/#std;debug.assert)
```

Comment:
This is a great spot to explain undefined behavior in zig and how it's caught in debug and release-safe, but optimized out of release-fast and release-small, there's a good explanation here https://github.com/ziglang/zig/blob/6fc822a9481c0d2c8b37f26f41be603dd77ab5a4/lib/std/debug.zig#L222

it's so simple that we might want to include it's implementation too.

I would also move the static assertion code up here and add a comptime assert example. (static_assert is also requires C11 iirc so that's important too)

Reply:
got it. `unreachable` to be introduced and more on `static_assert` & comptime assert